src/components/PDFView/index.js

```diff
@@ -204,6 +225,16 @@ class PDFView extends Component {
      */
     renderPage({index, style}) {
         const pageWidth = this.calculatePageWidth();
+        const pageHeight = this.calculatePageHeight(index);
+
+        // Each platform has a different default devicePixelRatio and different canvas limits, we need to verify that
+        // with the default devicePixelRatio it will be able to diplay the pdf correctly, if not we must change the devicePixelRatio.
+        const nbPixels = pageWidth * pageHeight;
+        const ratioHeight = this.props.maxCanvasHeight / pageHeight;
+        const ratioWidth = this.props.maxCanvasWidth / pageWidth;
+        const ratioArea = Math.sqrt(this.props.maxCanvasArea / nbPixels);
+        const ratio = Math.min(ratioHeight, ratioArea, ratioWidth);
+        const devicePixelRatio = ratio > window.devicePixelRatio ? undefined : ratio;
```

Comment:
This is re-calculated on every scroll and every render, even if this is a single-page document. 
We don't have to do the same calculations on every scroll since nothing has changed.
Let's use `_.memoize` for this, we can have a separate method called `this.getDevicePixelRatio`, we'd only need to pass `pageIndex` and `pageWidth`

Reply:
This is just 4-5 operations, it takes very short time, but I did it